rinja_parser/src/expr.rs

```diff
@@ -189,15 +189,24 @@ impl<'a> Expr<'a> {
         ))(i)
     }
 
-    expr_prec_layer!(or, and, "||");
-    expr_prec_layer!(and, compare, "&&");
-    expr_prec_layer!(compare, bor, "==", "!=", ">=", ">", "<=", "<");
-    expr_prec_layer!(bor, bxor, "|");
-    expr_prec_layer!(bxor, band, "^");
-    expr_prec_layer!(band, shifts, "&");
-    expr_prec_layer!(shifts, addsub, ">>", "<<");
-    expr_prec_layer!(addsub, muldivmod, "+", "-");
-    expr_prec_layer!(muldivmod, filtered, "*", "/", "%");
+    expr_prec_layer!(or, and, tag("||"));
+    expr_prec_layer!(and, compare, tag("&&"));
+    expr_prec_layer!(
+        compare,
+        bor,
+        tag("=="),
+        tag("!="),
+        tag(">="),
+        tag(">"),
+        tag("<="),
+        tag("<")
+    );
+    expr_prec_layer!(bor, bxor, value("|", keyword("bitor")));
+    expr_prec_layer!(bxor, band, alt((tag("^"), value("^", keyword("xor")))));
```

Comment:
Sorry I meant to emit an error in case the character is used instead of the keyword.

Reply:
You're totally right, changed.